pkg/ccl/crosscluster/logical/logical_replication_job_test.go

```diff
@@ -1101,6 +1235,50 @@ func CreateScatteredTable(t *testing.T, db *sqlutils.SQLRunner, numNodes int, db
 	}, timeout)
 }
 
+func GetPGURLs(
```

Comment:
what if you added this logic to `setupServerWithNumDBs` and returned a pgURLs array there?

Reply:
I think it would be nice, but `setupLogicalServer `also calls `setupServerWithNumDBs`. Propagating the cleanup function would require a lot more refactoring that would be better for a followup IMO. Unless you think I can just ignore the cleanup function since the test frees all resources anyway.

Reply:
i see, your point about cleanup makes sense. what if `setupServerWithNumDBs` returned a list of db names? My main hesitation with this func is that it constructs the dbnames out of thin air.

Reply:
Done, agreed that makes more sense